server/app.js

```diff
@@ -189,19 +189,19 @@ app.get('/user/:username', async (req, res, next)=>{
 				console.error(err);
 			});
 
-		for (const brew of brews) {
-			const match = googleBrews?.findIndex((b)=>b.editId === brew.editId) ?? -1;
-			if(match !== -1) {
-				brew.googleId = googleBrews[match].googleId;
-				brew.stubbed = true;
-				brew.pageCount = googleBrews[match].pageCount;
-				brew.renderer = googleBrews[match].renderer;
-				brew.version = googleBrews[match].version;
-				googleBrews.splice(match, 1);
+		if(googleBrews && googleBrews.length > 0) {
+			for (const brew of brews) {
+				const match = googleBrews.findIndex((b)=>b.editId === brew.editId);
+				if(match !== -1) {
+					brew.googleId = googleBrews[match].googleId;
+					brew.stubbed = true;
+					brew.pageCount = googleBrews[match].pageCount;
+					brew.renderer = googleBrews[match].renderer;
+					brew.version = googleBrews[match].version;
+					googleBrews.splice(match, 1);
+				}

```

Comment:
Sorry, for this I meant something like:

```
for (const brew of brews.filter(b=>b.googleId)) {  // pre-filter so we only loop over stubs
	if(!googleBrews?.length) //break out of the loop once we've reached all the googleBrews
		break;
	const match = googleBrews.findIndex((b)=>b.editId === brew.editId);
	if(match !== -1) {
		brew.googleId = googleBrews[match].googleId;
		brew.stubbed = true;
		brew.pageCount = googleBrews[match].pageCount;
		brew.renderer = googleBrews[match].renderer;
		brew.version = googleBrews[match].version;
		googleBrews.splice(match, 1);
	}
}
```

Reply:
Sorry I didn't notice the pre-filtering bit initially -- check it out now. And I actually moved the for loop into the if statement that was happening right after it, which validates the googleBrews exist and have a length of greater than one. Doesn't make sense to call the for loop at all if those two conditions aren't met.

Reply:
Ok I see it now. This looks fine. 👍